src/Parser/Core/Modules/DamageTaken.js

```diff
@@ -65,9 +65,44 @@ class DamageTaken extends Analyzer {
     return this._addDamage(ability, -amount, -absorbed, -overkill);
   }
 
+  getTooltip(){
+    const physical = (this._byMagicSchool[MAGIC_SCHOOLS.ids.PHYSICAL])?this._byMagicSchool[MAGIC_SCHOOLS.ids.PHYSICAL].effective : 0;
+    const magical = this.total.effective - physical;
+    return `Damage taken by type:
+    <ul>
+    <li><b>Physical</b>: ${formatThousands(physical)} (${formatPercentage(physical / this.total.effective)}%)</li>
+    <li><b>Magic</b>: ${formatThousands(magical)} (${formatPercentage(magical / this.total.effective)}%)</li>
+    </ul>
+    Damage taken by magic school:
+    <ul>
+      ${Object.keys(this._byMagicSchool)
+        .map(type =>{
+          if(this._byMagicSchool[type].effective === 0) return ``;
+          return `<li><b>${MAGIC_SCHOOLS.names[type] || 'Unknown'}</b>: ${formatThousands(this._byMagicSchool[type].effective)} (${formatPercentage(this._byMagicSchool[type].effective / this.total.effective)}%)</li>`;
+        })
+    .join('')}
+    </ul>
+    Click the bar to switch between simple and detailed mode.`;
+  }
+
+  handleClick(){
+    const barSimple = document.getElementById("damage-taken-bar-simple");
+    const barDetailed = document.getElementById("damage-taken-bar-detailed");
+    if((!barSimple || !barDetailed)) return; //return if one of the two bars isn't found.
+    const detailed = barDetailed.style.display.toLowerCase() !== "none";
+    if(detailed){
```

Comment:
Toggling DOM elements with React is usually done differently. With React the regular way to do this is to store the state of the element and only render the elements you want. If you're interested you can find a full tutorial here https://reactjs.org/docs/hello-world.html, I personally couldn't recommend React as a front-end library more.

It's important to note that our application is structured in a way that does not fully use the React way and mixes React approaches with plain ES6 classes. This causes a lot of the best practices to work slightly different and often require much poorer solutions, so while I will recommend a solution close to *the React way*, it's not entirely how this is normally done with React.

You really are in no way required to learn these things for future PRs, I'm just saying all this to try and explain my reasoning behind my suggestion below and maybe poke your interest.

As I said earlier *the React way* would be to only render the elements applicable. Assuming you have a property named `detailView` in the class that should show the detail view if `true` and the simple view if `false`, I'd suggest solving this with the following:

```jsx
handleClick(){
  this.detailView = !this.detailView;
}
```

```jsx
        footer={(
          <Wrapper>
            {this.detailView ? (
              <div onClick={() => this.handleClick()} className="statistic-bar">
                {Object.keys(this._byMagicSchool)
                  .map(type => {
                    const effective = this._byMagicSchool[type].effective;
                    if(effective === 0) return ``;
                    return (
                      <div
                        key={type}
                        className={`spell-school-${type}-bg`}
                        style={{ width: `${effective / this.total.effective * 100}%` }}
                        data-tip={tooltip}
                      />
                    );
                  })}
              </div>
            ) : (
              <div onClick={() => this.handleClick()} className="statistic-bar">
                {(() => {
                  const physical = (this._byMagicSchool[MAGIC_SCHOOLS.ids.PHYSICAL])?this._byMagicSchool[MAGIC_SCHOOLS.ids.PHYSICAL].effective : 0;
                  const magical = this.total.effective - physical;
                  const simplifiedValues = {};
                  simplifiedValues[MAGIC_SCHOOLS.ids.PHYSICAL] = physical;
                  simplifiedValues[MAGIC_SCHOOLS.ids.SHADOW] = magical; // use shadow as placeholder for general magic
                  return Object.keys(simplifiedValues).map(type => {
                    const effective = simplifiedValues[type];
                    return (
                      <div
                        key={type}
                        className={`spell-school-${type}-bg`}
                        style={{ width: `${effective / this.total.effective * 100}%` }}
                        data-tip={tooltip}
                      />
                    );
                  });
                })()
                }
              </div>
            )}
          </Wrapper>
        )}
```

Do note I didn't apply the other suggestions about the rendering of these bars.